### PR TITLE
PWGEM/PhotonMeson: MCV0 studies and cleanup

### PR DESCRIPTION
**Pull request by f3sch**
First commit is a cleanup commit (functionality stays the same):
 - change bool comparisons from & to && -> gets rid of a lot of anoying compiler warnings since these appeared in header files
 - moved one enum in the photon meson namespace to not pollute the global namespace
 - some minor changes to improve readability (mostly subjective and just where I spotted it)

I can drop this commit upon request, just these compiler warnings are _really_ annoying when I wanted to search for the ones I was responsible for :)

Second commit added the following for the MCV0 study:
 - add information of kinematics of propagated mcParticles
 - add information of the photon particle
 - add primary vertex information

### CODE REVIEW

File: PWGEM/PhotonMeson/Core/V0PhotonCut.h [hunk 1]
```diff
@@ -137,8 +138,8 @@ class V0PhotonCut : public TNamed
       }
     }
 
-    bool isTPConly_pos = !pos.hasITS() & pos.hasTPC() & !pos.hasTRD() & !pos.hasTOF();
-    bool isTPConly_ele = !ele.hasITS() & ele.hasTPC() & !ele.hasTRD() & !ele.hasTOF();
+    bool isTPConly_pos = isTPConlyTrack(pos);
+    bool isTPConly_ele = isTPConlyTrack(pos);
```
Comment (mhemmer-cern) [inline PWGEM/PhotonMeson/Core/V0PhotonCut.h:142]: should't this be 
`bool isTPConly_ele = isTPConlyTrack(ele);`

File: PWGEM/PhotonMeson/TableProducer/createPCM.cxx [hunk 2]
```diff
@@ -189,29 +191,29 @@ struct createPCM {
...
+    float dp = RecoDecay::dotProd(std::array{pxneg, pyneg, pzneg}, std::array{pxpos + pxneg, pypos + pyneg, pzpos + pzneg});
     return std::sqrt(RecoDecay::p2(pxneg, pyneg, pzneg) - dp * dp / momTot);
   }
 
   template <typename TTrack>
   bool reconstructV0(TTrack const& ele, TTrack const& pos)
   {
-    bool isITSonly_pos = pos.hasITS() & !pos.hasTPC();
-    bool isITSonly_ele = ele.hasITS() & !ele.hasTPC();
-    bool isTPConly_pos = !pos.hasITS() & pos.hasTPC();
```
Comment (dsekihat) [inline PWGEM/PhotonMeson/TableProducer/createPCM.cxx:211]: isTPConly here means that tracks that do not have ITS hits. Namely, TPConly, TPC-TOF, TPC-TRD, TPC-TRD-TOF should be accepted. I am sorry for confusion.

Comment (f3sch) [reply PWGEM/PhotonMeson/TableProducer/createPCM.cxx:211]: Thank you for clarifying.

**Review by mhemmer-cern — APPROVED** [review]
looks good

**Pull request merged**